src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs

```diff
@@ -1404,10 +1411,12 @@ S ComputeStartSet()
                         {
                             Debug.Assert(_left is not null && _right is not null);
                             S startSet = _left.GetStartSet();
+                            // To avoid deep recursion of trying to get the startset of right
+                            // just return True when left can be nullable
+                            // It is always correct to return True as the startset
```

Comment:
Does this change disable the startset deduction for patterns like `a?bR`? This should have a startset of `[ab]` but with the new change would have `.` if I understand this correctly, so wouldn't benefit from startset optimization. If so this piece of code is probably a target for an iterative rewrite.

Reply:
You are right, there should be a smarter check here. Not sure at this point what that would be.

Reply:
After some thought, I know how to fix this cleanly. I'll compute start-sets for all nodes at construction time. Similar to CoputeHashCode. This adds virtually no overhead and eliminates the recursion issue and is more efficient also. Startsets are used frequently so its seems not an overkill to add this extra info to each node.

Reply:
fixed